types/globals.d.ts

```diff
@@ -5,3 +5,7 @@ type DeepPartial<T> = {
     ? ReadonlyArray<DeepPartial<U>>
     : DeepPartial<T[P]>
 };
+
+// prettier-ignore
+type PromiseReturnType<T extends (...args: any[]) => any> =
```

Comment:
Could you explain the purpose of this?

Reply:
It unwraps the the resolved return value of a function that returns a promise.

See [this playground example](https://www.typescriptlang.org/play/index.html#src=type%20PromiseReturnType%3CT%20extends%20(...args%3A%20any%5B%5D)%20%3D%3E%20any%3E%20%3D%0D%0A%20%20ReturnType%3CT%3E%20extends%20Promise%3Cinfer%20U%3E%20%3F%20U%20%3A%20T%3B%0D%0A%0D%0Atype%20Foobar%20%3D%20()%20%3D%3E%20Promise%3C%7B%20foo%3A%20number%3B%20bar%3A%20number%20%7D%3E%3B%0D%0A%0D%0A%0D%0Adeclare%20const%20hover_this_const_to_see_the_type%3A%20PromiseReturnType%3CFoobar%3E) for a better idea.